PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = "hr.applicant"
-    _inherit = "hr.job"
-    _inherit = 'survey.survey'
-    _inherit = "survey.user_input"

Total Changes: 15

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_recruitment_survey/models/hr_applicant.py
+++ b/addons/hr_recruitment_survey/models/hr_applicant.py
@@ -5,8 +5,8 @@ from odoo import fields, models, _
 from odoo.exceptions import UserError
 
 
-class Applicant(models.Model):
-    _inherit = "hr.applicant"
+class HrApplicant(models.Model):
+    _inherit = ["hr.applicant"]
 
     survey_id = fields.Many2one('survey.survey', related='job_id.survey_id', string="Survey", readonly=True)
     response_ids = fields.One2many('survey.user_input', 'applicant_id', string="Responses")

--- a/addons/hr_recruitment_survey/models/hr_job.py
+++ b/addons/hr_recruitment_survey/models/hr_job.py
@@ -3,8 +3,8 @@
 from odoo import fields, models, _
 
 
-class Job(models.Model):
-    _inherit = "hr.job"
+class HrJob(models.Model):
+    _inherit = ["hr.job"]
 
     survey_id = fields.Many2one(
         'survey.survey', "Interview Form",

--- a/addons/hr_recruitment_survey/models/survey_survey.py
+++ b/addons/hr_recruitment_survey/models/survey_survey.py
@@ -2,8 +2,9 @@
 
 from odoo import api, fields, models
 
+
 class SurveySurvey(models.Model):
-    _inherit = 'survey.survey'
+    _inherit = ['survey.survey']
 
     survey_type = fields.Selection(selection_add=[('recruitment', 'Recruitment')], ondelete={'recruitment': 'set default'})
     hr_job_ids = fields.One2many("hr.job", "survey_id", string="Job Position")

--- a/addons/hr_recruitment_survey/models/survey_user_input.py
+++ b/addons/hr_recruitment_survey/models/survey_user_input.py
@@ -3,8 +3,8 @@
 from odoo import fields, models, _
 
 
-class SurveyUserInput(models.Model):
-    _inherit = "survey.user_input"
+class SurveyUser_Input(models.Model):
+    _inherit = ["survey.user_input"]
 
     applicant_id = fields.Many2one('hr.applicant', string='Applicant', index='btree_not_null')
 
